### PR TITLE
Amp local validation

### PR DESCRIPTION
**Pull request by gtrufitt**
## What does this change?

I added that developers should run  `make validate-amp` to the PR guidelines but then realised that just runs it on the live site, so added a local validation check to the tool to allow us to run it on our localhost urls.

@guardian/dotcom-platform

### CODE REVIEW

File: tools/amp-validation/index.js [hunk 1]
```diff
@@ -26,10 +24,12 @@ function checkEndpoints(devChannel) {
 }
 
 function runValidator(validator, devChannel) {
+    const isDev = process.argv[2] || false;
```
Comment (TBonnin) [inline tools/amp-validation/index.js]: not crucial but when I see `node index.js true` I have no idea what the second argument `true` means.
Can we make it more explicit? iex: `node index.js --dev`

File: tools/amp-validation/index.js [hunk 2]
```diff
@@ -26,10 +24,12 @@ function checkEndpoints(devChannel) {
...
+    
     return endpoint => fetchPage
-        .get(endpoint)
+        .get(endpoint, isDev)
         .then(res => {
-            console.log(`Checking the AMP validity (${devChannel ? 'pre-release' : 'release'}) of the page at ${endpoint}, result is:`);
+            console.log(`Checking the AMP validity (${devChannel ? 'pre-release' : 'release'}) of the page at ${isDev ? 'localhost:9000' : ''}${endpoint}, result is:`);
```
Comment (TBonnin) [inline tools/amp-validation/index.js]: Is it enough to run the `article` app locally or does the amp test require another app? 
Also how do we explain testers what needs to run locally?

File: tools/amp-validation/index.js (hunk 1 above)
Comment (desbo) [inline tools/amp-validation/index.js]: the convention is to use `NODE_ENV=dev` and `process.env`, I'd prefix the command with the env var

Comment (TBonnin) [inline tools/amp-validation/index.js]: works for me

**Comment by gtrufitt** [comment]
Thanks both, I'm now using node env and showing a big message telling people they should run dev-build or article

**Review by desbo — APPROVED** [review]
approved-29149_960_720

**Comment by gtrufitt** [comment]
THANKS

**Comment by johnduffell** [comment]
that's massive!

**Comment by jfsoul** [comment]
Nice idea for local testing 👏 

Could have always used an npm script for `NODE_ENV=dev node index.js` (there is a `start` script already) and decoupled slightly more, but not worth worrying about now.

**Pull request merged**